transcom/mymove
# B 17615 weight ticket designator for payment request uploads

## [B-17615](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A844121&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

Added a requirement of the Prime to add a parameter of `isWeightTicket` to a POST request to the `/createUpload` endpoint. 

This required adding a migration file to add a column, `isWeightTicket` to the `proof_of_service_docs` table. Made the default value of `FALSE` to avoid any `NULL` panics.

Updated the Go Struct for `ProofOfServiceDocs` model to include the `isWeightTicket` boolean value.

Updated the payload that is returned for an upload if the upload is:
1. A proof of service doc
2. A proof of service doc that is a weight ticket
This adds an `isWeightTicket` key to the upload array that is used by the UI when reviewing payment requests and their respective documents.

*NOTE* This story did not have criteria to update the Prime simulator, so when using the Prime Sim you will be unable to test this or add a `isWeightTicket` value to the upload.

In order to see this change in the frontend, I also had to update the `model_to_payload.go` file for the GHC API and allow including the parameter in a `GET` request for payment requests.

*DON'T FREAK OUT* by the number of "Files changed" in this request. When changing any property in `swagger-def`, it changes several auto-generated files. You good. We good. It's all good.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Log into MilMove as the Prime
2. Using Postman, send a POST request to `/createUpload` on an existing payment request. Ensure your body is in `formData` as the `Content-Type` and send it.
```
{
   "file": "filePath",
   "isWeightTicket": true
}
```
3. Sign in as an office user and perform a `GET` request to `/payment-requests/{paymentRequestID}`
4. Confirm you can see the `isWeightTicket` parameter for the `proofOfServiceDoc` and `upload` you added to the payment request
